nrf-hal-common/src/spim.rs

```diff
@@ -157,67 +166,89 @@ where
             // there.
             unsafe { w.orc().bits(orc) });
 
-        Spim(spim)
+        Spim {
+            periph: spim,
+            pins,
+        }
     }
 
     /// Internal helper function to setup and execute SPIM DMA transfer.
     fn do_spi_dma_transfer(&mut self, tx: DmaSlice, rx: DmaSlice) -> Result<(), Error> {
+        self.start_spi_dma_transfer(&tx, &rx);
+
+        // Wait for END event.
+        //
+        // This event is triggered once both transmitting and receiving are
+        // done.
+        while !self.is_spi_dma_transfer_complete() {}
```

Comment:
Is there an alternative to spinning here: could we setup an interrupt and go to sleep? I'm not an expert at all so apologies if this is a silly suggestion.